src/Data/Array/NonEmpty.purs

```diff
@@ -361,6 +364,43 @@ fold1 = F.fold1
 intercalate :: forall a. Semigroup a => a -> NonEmptyArray a -> a
 intercalate = F.intercalate
 
+-- | The 'transpose' function transposes the rows and columns of its argument.
+-- | For example,
+-- |
+-- | ```purescript
+-- | transpose 
+-- |   (NonEmptyArray [ NonEmptyArray [1, 2, 3]
+-- |                  , NonEmptyArray [4, 5, 6]
+-- |                  ]) == 
+-- |   (NonEmptyArray [ NonEmptyArray [1, 4]
+-- |                  , NonEmptyArray [2, 5]
+-- |                  , NonEmptyArray [3, 6]
+-- |                  ])
+-- | ```
+-- |
+-- | If some of the rows are shorter than the following rows, their elements are skipped:
+-- |
+-- | ```purescript
+-- | transpose 
+-- |   (NonEmptyArray [ NonEmptyArray [10, 11]
+-- |                  , NonEmptyArray [20]
+-- |                  , NonEmptyArray [30, 31, 32]
+-- |                  ]) == 
+-- |   (NomEmptyArray [ NonEmptyArray [10, 20, 30]
+-- |                  , NonEmptyArray [11, 31]
+-- |                  , NonEmptyArray [32]
+-- |                  ])
+-- | ```
+transpose :: forall a. NonEmptyArray (NonEmptyArray a) -> NonEmptyArray (NonEmptyArray a)
+transpose = 
+  (coerce :: (Array (Array a)) -> (NonEmptyArray (NonEmptyArray a))) 
+    <<< A.transpose <<< coerce
+
+-- | `transpose`' is identical to `transpose` other than that the inner arrays are each
+-- | a standard `Array` and not a `NonEmptyArray`.
+transpose' :: forall a. NonEmptyArray (Array a) -> NonEmptyArray (Array a)
+transpose' = unsafeAdapt A.transpose
+
```

Comment:
@garyb Just another thought, but should be there be a `transpose` variant when the input is `Array (NonEmptyArray a)`?

Reply:
I did give that a moment's thought, but I figure we can skip it in favour of doing like `maybe [] (NEA.toArray <<< NEA.transpose) (NEA.fromArray xs)` if it arises.

Reply:
Ah, good point!